### PR TITLE
[WIP] Copy refs only on publish

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/dotnet/cli/issues/616

### CODE REVIEW

**Comment by anurse** [comment]
If the OSX Debug Build fails again, please wait for my confirmation before restarting it. I'm going to try taking a look at the Workspace and see why this failure is happening

**Comment by anurse** [comment]
Ok, I've captured a supposedly broken lock file. You can resume re-running OSX Debug :)

**Comment by pakrym** [comment]
@dotnet-bot test this please

**Comment by pakrym** [comment]
@dotnet-bot test this please

File: src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs [hunk 1]
```diff
@@ -24,24 +29,126 @@ public IEnumerable<string> ResolveReferencePaths()
         {
             var entryAssembly = _entryAssembly.Value;
             var entryAssemblyName = entryAssembly.GetName().Name;
-            var basePath = GetRefsLocation();
 
-            foreach (var assembly in Assemblies)
+            string basePath;
+            string fullName;
+
+            var appBase = Path.GetDirectoryName(entryAssembly.Location);
```
Comment (davidfowl) [inline src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs]: We should use `AppContext.ApplicationBasePath`

Comment (pakrym) [inline src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs]: It would be the same as `entryAssembly.Location ` or could be changed?

Comment (pakrym) [inline src/Microsoft.Extensions.DependencyModel/CompilationLibrary.cs]: AppContext is only in .net 4.6

**Comment by moozzyk** [comment]
@davidfowl - without this MVC is currently broken - if you publish the project it won't work.

**Pull request merged**